setup.py

```diff
@@ -2,8 +2,89 @@
 # -*- encoding: utf-8 -*-
 
 from setuptools import setup
+from distutils.command import build
+from distutils.command import install
+#from distutils.command import install_scripts
+import distutils
+import subprocess
+
 from kiwi.version import __version__
 
+
+class kiwi_build(build.build):
+    """
+    Custom build command
+    """
+    build.build.user_options += [
+        ('cflags=', None, 'compile flags')
+    ]
+
+    def initialize_options(self):
+        """
+        Set default values for options
+        Each user option must be listed here with their default value.
+        """
+        build.build.initialize_options(self)
+        self.cflags = ''
+
+    def run(self):
+        """
+        Run first the related KIWI C compilation and after that
+        the usual Python build
+        """
+        # kiwi C tools compilation
+        command = ['make']
+        if self.cflags:
+            command.append('CFLAGS=%s' % self.cflags)
+        command.append('tools')
+        self.announce(
+            'Running make tools target: %s' % str(command),
+            level=distutils.log.INFO
+        )
+        self.announce(
+            subprocess.check_output(command).decode(),
+            level=distutils.log.INFO
+        )
+
+        # standard build process
+        build.build.run(self)
+
+
+class kiwi_install(install.install):
+    """
+    Custom install command
+    """
+    sub_commands = [
+        ('install_lib', lambda self:True),
+        ('install_headers', lambda self:False),
+        ('install_scripts', lambda self:True),
+        ('install_data', lambda self:False),
+        ('install_egg_info', lambda self:True),
+    ]
+
+    def run(self):
```

Comment:
I'm not sure if this is a problem or you want to support that use case. I tried to install the package in a different location which is allowed with the "install" subcommand. I tried this:

```
$ ./setup.py install --prefix /tmp/kiwi
running install
Running make install target: ['make', 'install']
install: cannot stat ‘kversion’: No such file or directory
install: cannot stat ‘utimer’: No such file or directory
install: cannot stat ‘dcounter’: No such file or directory
install: cannot stat ‘startshell’: No such file or directory
install: cannot stat ‘isconsole’: No such file or directory
install: cannot stat ‘driveready’: No such file or directory
install: cannot stat ‘setctsid’: No such file or directory
make[1]: *** [install] Error 1
make: *** [install] Error 2
Traceback (most recent call last):
  File "./setup.py", line 130, in <module>
    setup(**config)
  File "/usr/lib64/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/lib64/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/lib64/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "./setup.py", line 80, in run
    subprocess.check_output(command).decode(),
  File "/usr/lib64/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['make', 'install']' returned non-zero exit status 2
```